test/Microsoft.Azure.Devices.Edge.Test.Common/config/EdgeConfiguration.cs

```diff
@@ -29,6 +30,30 @@ public class EdgeConfiguration
                 .ToArray();
         }
 
+        public static EdgeConfiguration Create(string deviceId, List<ModuleConfiguration> modules)
+        {
+            var names = modules.Select(m => new StringBuilder(m.Name).ToString()).ToArray();
+            var images = modules.Select(m => new StringBuilder(m.Image).ToString()).ToArray();
```

Comment:
No. Since strings are immutable, this isn't really a concern...updated references will point to a new string, other references will still point to the old. I'll simplify this.

Reply:
Fixed.